nix/modules/uki.nix

```diff
@@ -0,0 +1,56 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.boot.loader.uki;
+in
+{
+  options.boot.loader.uki = {
+    enable = lib.mkEnableOption "UKI";
+
+    stub = lib.mkOption {
+      type = lib.types.path;
+      description = "Path to the UKI stub to use.";
+    };
+  };
+
+  config =
+    let
+      systemdUkify = pkgs.systemdMinimal.override {
+        withEfi = true;
+        withUkify = true;
+      };
+    in
+    lib.mkIf cfg.enable {
+      boot.bootspec.enable = true;
+      boot.loader.external = {
+        enable = true;
+        installHook =
+          let
+            runtimeInputs = with pkgs; [ jq systemd binutils ];
+            bootspecNamespace = ''"org.nixos.bootspec.v1"'';
+          in
+          pkgs.writeShellScript "install-uki" ''
+            set -euo pipefail
+            export PATH="${lib.makeBinPath runtimeInputs}:$PATH"
+
+            boot_json=/nix/var/nix/profiles/system-1-link/boot.json
```

Comment:
that's not prod ready :D

Reply:
Well, since we can't really rebuild properly in a VM test anyway its as prod ready as we need for the tests :D